libs/components/tiles/src/lib/modules/tiles/tile-dashboard/tile-dashboard.service.ts

```diff
@@ -45,28 +47,38 @@ export class SkyTileDashboardService {
   @Output()
   public dashboardInitialized = new EventEmitter<void>();
 
-  private tileComponents: ComponentRef<any>[];
-
-  private config: SkyTileDashboardConfig;
-
-  private defaultConfig: SkyTileDashboardConfig;
-
-  private columns: QueryList<SkyTileDashboardColumnComponent>;
-
-  private singleColumn: SkyTileDashboardColumnComponent;
-
-  private mediaSubscription: Subscription;
-
-  private settingsKey: string;
-
+  #columns: QueryList<SkyTileDashboardColumnComponent> | undefined;
+  #config: SkyTileDashboardConfig | undefined;
+  #dragulaService: DragulaService;
+  #dynamicComponentService: SkyDynamicComponentService;
+  #mediaSubscription: Subscription | undefined;
+  #mediaQuery: SkyMediaQueryService;
+  #singleColumn: SkyTileDashboardColumnComponent | undefined;
+  #settingsKey: string | undefined;
+  #tileComponents: ComponentRef<any>[] | undefined;
+  #uiConfigService: SkyUIConfigService;
+
+  // TODO: remove @optional tag in a future breaking change
   constructor(
-    private dragulaService: DragulaService,
-    private mediaQuery: SkyMediaQueryService,
-    private uiConfigService: SkyUIConfigService
+    dragulaService: DragulaService,
+    mediaQuery: SkyMediaQueryService,
+    uiConfigService: SkyUIConfigService,
+    @Optional() dynamicComponentService?: SkyDynamicComponentService
   ) {
-    this.bagId = 'sky-tile-dashboard-bag-' + ++bagIdIndex;
+    if (!dynamicComponentService) {
+      console.warn(
+        'The SkyTileDashboardService was created without a reference to the SkyDynamicComponentService. The service will be unable to create tiles from the dashboard configuration. Constructing the SkyTileDashboardService without a SkyDynamicComponentService will be removed in a future breaking change.'
+      );
+    }
 
-    this.initDragula();
+    this.#dragulaService = dragulaService;
+    this.#dynamicComponentService = dynamicComponentService!;
```

Comment:
Let's remove the non-null, here, and add an `if` check when we use the service, below, so that our source code reflects the optional nature of the property's signature.
```
/* istanbul ignore else */
if (tile && this.#dynamicComponentService) {
```

Reply:
Updated